osutil/cmdline.go

```diff
@@ -157,3 +173,38 @@ func KernelCommandLineSplit(s string) (out []string, err error) {
 	}
 	return out, nil
 }
+
+// KernelCommandLineKeyValue returns a map of the specified keys to the values
+// set for them in the kernel command line. If the value is missing from the
+// kernel command line, the key is omitted from the returned map.
+func KernelCommandLineKeyValue(keys ...string) (map[string]string, error) {
+	buf, err := ioutil.ReadFile(procCmdline)
+	if err != nil {
+		return nil, err
+	}
+	params, err := KernelCommandLineSplit(strings.TrimSpace(string(buf)))
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string]string, len(keys))
+
+	for _, param := range params {
+		for _, key := range keys {
+			if strings.HasPrefix(param, fmt.Sprintf("%s=", key)) {
+				res := strings.SplitN(param, "=", 2)
+				if len(res) == 2 {
+					m[key] = res[1]
+				} else {
+					// length not of N means the empty string was passed, so set
+					// that, so that callers can tell whether a parameter was not
+					// set at all (by checking if the key is in the returned map)
+					// or if it was literally set to the empty string (it's there
+					// and set to the empty string)
+					m[key] = ""
```

Comment:
this case doesn't see to be tested

Reply:
AFAICT this branch is impossible to enter because of the earlier `key=` prefix check.

Reply:
yes,  because  SplitN("a=", "=", 2) returns ["a", ""].  So I think the if should just go away